### PR TITLE
MODPATBLK-62 add mapping for FeeFineTypeId

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Approach
- fetch FeeFineTypeId and map it to event;
- fix tests;

Resolves: https://issues.folio.org/browse/MODPATBLK-62

### CODE REVIEW

File: src/test/java/org/folio/rest/impl/SynchronizationAPITests.java [hunk 1]
```diff
@@ -357,6 +390,19 @@ private static String makeAccountsResponseBody() {
       .encodePrettily();
   }
 
+  private static String makeFeeFinesResponseBody() {
```
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply src/test/java/org/folio/rest/impl/SynchronizationAPITests.java:393]: done

File: src/main/java/org/folio/service/FeesFinesEventsGenerationService.java [hunk 2]
```diff
@@ -60,16 +62,18 @@ protected void addGeneratedEventsForEachPagesToList(SynchronizationJob syncJob,
...
-      .map(this::generateFeeFineBalanceChangedEvent)
-      .reduce(Future.succeededFuture(), (a, b) -> a.compose(r -> b));
+    return feeFineOkapiClient.fetchFeeFineTypes()
+      .compose(feeFineTypes -> records.stream()
+        .map(account -> generateFeeFineBalanceChangedEvent(account, feeFineTypes))
+        .reduce(Future.succeededFuture(), (a, b) -> a.compose(r -> b)));
   }
 
-  private Future<String> generateFeeFineBalanceChangedEvent(Account account) {
+  private Future<String> generateFeeFineBalanceChangedEvent(Account account, Map<String, String> eventTypes) {
```
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply src/main/java/org/folio/service/FeesFinesEventsGenerationService.java]: updated

File: src/main/java/org/folio/rest/client/FeesFinesOkapiClient.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+
+public class FeesFinesOkapiClient extends OkapiClient {
+
+  public FeesFinesOkapiClient(Vertx vertx, Map<String, String> okapiHeaders) {
+    super(vertx, okapiHeaders);
+  }
+
+  public Future<Map<String, String>> fetchFeeFineTypes() {
```
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply src/main/java/org/folio/rest/client/FeesFinesOkapiClient.java]: done

File: src/main/java/org/folio/rest/client/FeesFinesOkapiClient.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.folio.rest.jaxrs.model.FeeFineCollection;
+import org.folio.rest.jaxrs.model.Feefine;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+
+public class FeesFinesOkapiClient extends OkapiClient {
```
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply src/main/java/org/folio/rest/client/FeesFinesOkapiClient.java]: renamed

File: ramls/automated-patron-blocks.raml [hunk 5]
```diff
@@ -14,6 +14,8 @@ types:
   account: !include accountdata.json
   errors: !include raml-util/schemas/errors.schema
   automatedPatronBlocks: !include automated-patron-blocks.json
+  feeFine: !include feefinedata.json
```
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply ramls/automated-patron-blocks.raml]: renamed

File: src/test/java/org/folio/rest/impl/SynchronizationAPITests.java [hunk 6]
```diff
@@ -357,6 +366,19 @@ private static String makeAccountsResponseBody() {
...
+    JsonObject feeFine = new JsonObject()
+      .put("id", randomId())
```
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply src/test/java/org/folio/rest/impl/SynchronizationAPITests.java]: added

File: src/main/java/org/folio/service/FeesFinesEventsGenerationService.java [hunk 7]
```diff
@@ -60,16 +61,31 @@ protected void addGeneratedEventsForEachPagesToList(SynchronizationJob syncJob,
...
-      .reduce(Future.succeededFuture(), (a, b) -> a.compose(r -> b));
+    return fetchFeeFineTypes()
+      .compose(feeFineTypes -> records.stream()
+        .map(account -> generateFeeFineBalanceChangedEvent(account, feeFineTypes))
+        .reduce(Future.succeededFuture(), (a, b) -> a.compose(r -> b)));
   }
 
-  private Future<String> generateFeeFineBalanceChangedEvent(Account account) {
+  private Future<Map<String, String>> fetchFeeFineTypes() {
+    return okapiClient.getMany("/feefines", FEES_FINES_LIMIT, 0)
```
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply src/main/java/org/folio/service/FeesFinesEventsGenerationService.java]: done

File: src/main/java/org/folio/service/FeesFinesEventsGenerationService.java [hunk 8]
```diff
@@ -1,13 +1,13 @@
...
 
 import static io.vertx.core.Future.failedFuture;
-import static io.vertx.core.Future.succeededFuture;
 
 import java.math.BigDecimal;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
+import org.apache.commons.collections4.map.HashedMap;
```
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply src/main/java/org/folio/service/FeesFinesEventsGenerationService.java]: removed

**Review by OleksandrVidinieiev — APPROVED** [review]

**Pull request merged**